### PR TITLE
Set Purpur as upstream

### PR DESCRIPTION
**Pull request by kugge**
Solves #314
Deletes non-canonical patches (https://pastebin.com/BwThixJR) and set Purpur as upstream for MP-Server.
 
Why this PR:
- Get rid of external patches maintenance
- Delete airplaine config
- Include all pufferfish patches and purpur features
- Always up to date purpur/pufferfish patches

Why Purpur:
- Always up to date paper version (pufferfish only updates every 2 weeks..)
- Include better debugging tools by default
- Include great optimisations/features/api for servers
- Does not promote paid software

Why not Paper:
Multipaper community want to include Pufferfish optimisations and they're right. I think this is the best way to do it. Less useless copy-paste maintenance work, great features and always up to date thanks to Purpur. Also it get rid of old code (Airplane).

### CODE REVIEW

**Comment by xymb-endcrystalme** [comment]
I agree that changing to Purpur may be a good idea.

**Comment by ham1255** [comment]
so pufferfish code still included?

**Comment by kugge** [comment]
> so pufferfish code still included?, since i see no trace for them.

Its included in Purpur and maintained by Purpur

https://github.com/PurpurMC/Purpur/blob/ver/1.19.3/patches/server/0001-Pufferfish-Server-Changes.patch

**Review by ham1255 — APPROVED** [review]

**Comment by KaspianDev** [comment]
Purpur has some changes I personally don't like. Like fully removing timings.

File: build.gradle.kts [hunk 1]
```diff
@@ -59,37 +59,42 @@ paperweight {
     serverProject.set(project(":MultiPaper-Server"))
 
     remapRepo.set("https://maven.fabricmc.net/")
-    decompileRepo.set("https://files.minecraftforge.net/maven/")
+    decompileRepo.set("https://maven.quiltmc.org")
```
[the remark this answers is not in the document]
Comment (kugge) [reply build.gradle.kts:62]: That was just for quiltmc web interface. Didn't make any difference, should i revert it ?

**Comment by kugge** [comment]
> Purpur has some changes I personally don't like. Like fully removing timings.

You can always activate them in config.

**Comment by ham1255** [comment]
spark is already included which good?

**Comment by KaspianDev** [comment]
> > Purpur has some changes I personally don't like. Like fully removing timings.
> 
> You can always activate them in config.

Are you sure?
https://github.com/PurpurMC/Purpur/blob/ver/1.19.3/patches/server/0278-Remove-Timings.patch#L19
It's commented out.

> spark is already included which good?
Spark is harder to understand for average user. And timings are known by people for years.

**Comment by ham1255** [comment]
> > Purpur has some changes I personally don't like. Like fully removing timings.
> 
> You can always activate them in config.

maybe look into dropping that patch?

**Comment by kugge** [comment]
> > > Purpur has some changes I personally don't like. Like fully removing timings.
> > 
> > 
> > You can always activate them in config.
> 
> maybe look into dropping that patch?

Made a patch that revert Purpur "remove timings" patch.

But it is a good idea ?
According to Billy Galbreath: 

```
Timings runs all the time. Even if you disable it, all those hundreds of thousands of noop calls per second are still taking up processing power. If all you need is /tps then you'll be glad to know that Spark does not use any of your CPU while it's not being used, plus it comes with a nicer and more accurate /tps command.

Unfortunately, with the nature of the problem with Timings eating up CPU time even when it's disabled, the only solution does not permit us to add a way to simply toggle it back on. We've been wanting to remove Timings for years now, but we've held off until Spark was fully capable of replacing it. We're not going to just remove Timings (no matter how bad it is) without giving you guys a good replacement.
```
Plus, it seems that there is no reason to use timings over spark.

**Comment by xymb-endcrystalme** [comment]
I'm all for removing timings. They are slow (they slow down execution) and are obsoleted by Spark plugin and it's glorious flame chart.

**👍 for removing timings!**

**Comment by ham1255** [comment]
> > > > Purpur has some changes I personally don't like. Like fully removing timings.
> > > 
> > > 
> > > You can always activate them in config.
> > 
> > 
> > maybe look into dropping that patch?
> 
> Made a patch that revert Purpur "remove timings" patch.
> 
> But it is a good idea ? According to Billy Galbreath:
> 
> > Timings runs all the time. Even if you disable it, all those hundreds of thousands of noop calls per second are still taking up processing power. If all you need is /tps then you'll be glad to know that Spark does not use any of your CPU while it's not being used, plus it comes with a nicer and more accurate /tps command.
> > Unfortunately, with the nature of the problem with Timings eating up CPU time even when it's disabled, the only solution does not permit us to add a way to simply toggle it back on. We've been wanting to remove Timings for years now, but we've held off until Spark was fully capable of replacing it. We're not going to just remove Timings (no matter how bad it is) without giving you guys a good replacement.
> 
> Plus, it seems that there is no reason to use timings over spark.

Sounds reasonable then

File: build.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (PureGero) [reply build.gradle.kts:62]: This should probably be set the same as Purpur. `remapRepo` too.
```https://repo.papermc.io/repository/maven-public/```
https://github.com/PurpurMC/Purpur/blob/ver/1.19.3/build.gradle.kts#L19

[the remark this answers is not in the document]
Comment (kugge) [reply build.gradle.kts]: Done, using `https://repo.papermc.io/repository/maven-public/` as remap/decompile repo

**Pull request merged**